### PR TITLE
[sqlpp11] Fix libmariadb linkage, generate usage

### PR DESCRIPTION
**Pull request by dg0yt**

### CODE REVIEW

**Review by WangWeiLin-MV — APPROVED** [review]
The port features usage tests pass with the following triplets:

* x64-linux

File: ports/sqlpp11/portfile.cmake [hunk 1]
```diff
@@ -32,4 +32,16 @@ vcpkg_cmake_configure(
...
+if(FEATURES STREQUAL "core")
+    set(usage "This build of sqlpp11 doesn't include any connector.\n(Available via features: sqlite3, mariadb, mysql, postgresql.)\n")
+endif()
+foreach(component IN ITEMS SQLite3 SQLCipher MySQL MariaDB PostgreSQL)
+    string(TOLOWER "${component}" lib)
+    if("${lib}" IN_LIST FEATURES)
+        string(APPEND usage "\n  find_package(Sqlpp11 CONFIG REQUIRED COMPONENTS ${component})\n  target_link_libraries(main PRIVATE sqlpp11::${lib})\n")
+    endif()
+endforeach()
+file(WRITE "${CURRENT_PACKAGES_DIR}/share/${PORT}/usage" "${usage}")
```
Comment (JavierMatosD) [inline ports/sqlpp11/portfile.cmake:45]: Why do this as opposed to simply writing a usage file?

**Review by JavierMatosD — COMMENTED** [review]
This LGTM, except the usage instructions. The usage is supposed to be the minimum instructions to get a user running. This looks like it's trying to supply the whole thing.

File: ports/sqlpp11/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/sqlpp11/portfile.cmake:45]: For copy&paste and good UX with minimal output, subject to actual features. Both CMake commands should name the backend.

**Pull request merged**